### ISSUE TITLE
Curl error 3: URL rejected

### ISSUE THREAD

**Issue by Nomad1108** [GlitchEnzo/NuGetForUnity#656]
## Description

I get one error "Curl error 3: URL rejected: Error" per package when opening the NuGet for Unity window. Packages icons do not load, so best case scenario that's a cosmetic issue, but still annoying...

<img width="825" alt="Capture d’écran 2024-06-20 à 13 05 09" src="https://github.com/GlitchEnzo/NuGetForUnity/assets/59485866/da8d7a36-1009-4120-81fe-a4b4dafaa354">

-   **NuGetForUnity Version:** 4.1.1
-   **Unity Version:** Unity 2022.3.8f1
-   **Operating System:** MacOS

**Comment by Nomad1108** [comment]
I tried both plugin install methods: as GIT dependency via Package Manager and then via .unitypackage file but still got the issue.

I've also tried to change the placement to "Custom within Assets" but it did not solve the issue.

<img width="664" alt="Capture d’écran 2024-06-20 à 13 09 01" src="https://github.com/GlitchEnzo/NuGetForUnity/assets/59485866/2599c0cd-297a-46f8-892e-ebe35df932f9">

**Comment by DilaverSerif** [comment]
same

**Comment by igor84** [comment]
I tried it on a Mac and I couldn't reproduce the issue.  Can you open the icon URL in the browser (for example https://api.nuget.org/v3-flatcontainer/newtonsoft.json/13.0.3/icon)?

**Comment by DilaverSerif** [comment]
its okey but i have still errors. 
<img width="790" alt="Ekran Resmi 2024-09-11 10 57 12" src="https://github.com/user-attachments/assets/a73269a6-04ce-4d58-b857-bdb8ab2fe2b1">

**Comment by owlvark** [comment]
I was also getting those Curl errors in the console on a fresh install.  I went into the Preferences and checked "Use Verbose Logging" and restarted Unity and the Curl errors did not appear.  Perhaps the default `NuGet.config` has problems?

For reference, after the logging setting change these lines were added to the config:

`<?xml version="1.0" encoding="utf-8"?>`
`<packageSourceCredentials />`
`<add key="verbose" value="true" />`

and `<clear />` was removed.

**Comment by owlvark** [comment]
Ok, after further testing, I think it happens when there are spaces in the URL?

**Comment by JoC0de** [comment]
so they are only for File-URLs? Or how can a URL have spaces normally they are encoded with `%20`. Do you have spaces inside one of the folders that contain your Unity project?

**Comment by owlvark** [comment]
Yes, it is a file and there are spaces; it is trying to open a URL like this:

`file:////var/folders/vj/5sqv1srj7ds8y8ztqddggkww0000gn/T/Studio Name/Game Name/ER4xRB-z5RmCSiBeUlEktw`

This results in the Curl error and a log message that says `Web error: Malformed URL` towards the end.
In my tests, when the spaces in that URL string are replaced with `%20` the file loads as intended.

**Comment by Nomad1108** [comment]
I can confirm that my "Company Name" in Player Settings is two words with a space between them, which would add at least one space to the URL.

**Comment by ndnomad** [comment]
I am having this same error. I removed all the spaces in the offending paths, except one. Which is my Unity Projects folder, which hold my Project Name folders. Do I really have to rename all of my core Unity project folders to avoid this error?? I would hope this is a very simple bug for you to fix.

**Comment by JoC0de** [comment]
@ndnomad please check if the bug is resolved by the latest release.

**Comment by owlvark** [comment]
I tried out release v4.2.0 but the issue remains.  Using `new Uri` unfortunately does not encode spaces.
If it is hard to reproduce, here is a test case (MacOS):

```
var url = "file:////var/folders/space test";
var request = UnityWebRequest.Get(url);
var downloadHandler = new DownloadHandlerTexture(false);
request.downloadHandler = downloadHandler;
var operation = request.SendWebRequest();
```

The above code throws the "Curl error 3" exception. If the space in the file URL is replaced with `%20` it throws "Curl error 37: Couldn't open file" instead.

I think, if we want to avoid the controversy of [Uri.EscapeUriString](https://learn.microsoft.com/en-us/dotnet/api/system.uri.escapeuristring?view=net-9.0), in this case we could use `url.Replace(" ", "%20")`.
